tests/components/blue_current/__init__.py

```diff
@@ -1,52 +1,94 @@
 """Tests for the Blue Current integration."""
 from __future__ import annotations
 
-from unittest.mock import patch
+from asyncio import Event, Future
+from unittest.mock import MagicMock, patch
 
 from bluecurrent_api import Client
 
-from homeassistant.components.blue_current import DOMAIN, Connector
-from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.dispatcher import async_dispatcher_send
 
 from tests.common import MockConfigEntry
 
+DEFAULT_CHARGE_POINT = {
+    "evse_id": "101",
+    "model_type": "",
+    "name": "",
+}
+
+
+def create_client_mock(
+    loop_future: Future | None,
+    charge_point: dict,
+    status: dict | None,
+    grid: dict | None,
+) -> MagicMock:
+    """Create a mock of the bluecurrent-api Client."""
+    client_mock = MagicMock(spec=Client)
+    client_mock.started_loop = Event()
```

Comment:
The `started_loop` event could also be created in a fixture similar as the future.

Reply:
I thought about it but the event is always needed. I could also remove it from mock_client and return it from create_client and init_integration to access it in the test instead.

Reply:
I think it's more clear to have it separate from the mock client since it's not part of a real client.

Using functions or fixtures depends on the use case and test needs. You can mix and match that as you like.